aioasuswrt/asuswrt.py

```diff
@@ -255,18 +258,31 @@ def __init__(
         else:
             self.connection = SshConnection(host, port, username, password, ssh_key)
 
-    async def async_get_nvram(self, to_get):
+    async def async_get_nvram(self, to_get, use_cache=True):
         """Gets nvram"""
         data = {}
-        if to_get in GET_LIST:
+        if not (to_get in GET_LIST):
+            return data
+
+        now = datetime.utcnow()
+        if (
+            use_cache
+            and self._nvram_cache_timer
+            and self._cache_time > (now - self._nvram_cache_timer).total_seconds()
+        ):
+            lines = self._nvram_cache
+        else:
             lines = await self.connection.async_run_command(_NVRAM_CMD)
-            for item in GET_LIST[to_get]:
-                regex = rf"{item}=([\w.\-/: ]+)"
-                for line in lines:
-                    result = re.findall(regex, line)
-                    if result:
-                        data[item] = result[0]
-                        break
+            self._nvram_cache = lines
+            self._nvram_cache_timer = now
+
+        for item in GET_LIST[to_get]:
+            regex = rf"^{item}=([\w.\-/: ]+)"
+            for line in lines:
+                result = re.findall(regex, line)
+                if result:
+                    data[item] = result[0]
```

Comment:
Does it not make more sense to cache the data dict instead of the lines. That way we also only need to do this once for every cache hit.

Reply:
No, because the lines returned every call can be different and depend on `to_get` function parameter

Reply:
ahh yeah should have seen that